app_feup/lib/view/Widgets/exam_card.dart

```diff
@@ -85,7 +85,10 @@ class ExamCard extends GenericCard {
           end: exam.end,
           type: exam.examType,
         ),
-      )),
+       ),
+         color: (exam.examType.contains('''EN''')) || (exam.examType.contains('''MT'''))
+             ? null : Theme.of(context).highlightColor,
+       ),
```

Comment:
Since this logic is the same as in the lines 115-116, we should probably extract this into a method, wdyt?

Furthermore, I don't think we should return null for the color, even if the result will not change, just to make it more understandable what's happening.

Reply:
I created a function, examColor, in exam.dart that returns a color (backgroundColor or highlightColor) based on exam's type and context of widget. Also, implemented this function in exam_card.dart. What do you think?

Reply:
An entity in the model should do its best to avoid dealing with any logic that pertains to what the user is viewing. Can you move that method to the `exam_card` as well?

Reply:
Thank you!! Done 😀